code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -530,17 +551,26 @@ var/global/list/damage_icon_parts = list()
 	if(update_icons)   update_icons()
 
 /mob/living/carbon/human/update_inv_gloves(var/update_icons=1)
-	if(gloves)
+	overlays_standing[GLOVES_LAYER]	= null

```

Comment:
Everything is nulled out before the slot is written to, that's standard practise so that if something has just been removed, the slot will properly be drawn empty